include/oneapi/tbb/detail/_flow_graph_node_impl.h

```diff
@@ -647,6 +647,18 @@ struct emit_element {
         check_task_and_spawn(g, last_task);
         return emit_element<N-1>::emit_this(g,t,p);
     }
+
+#if __TBB_PREVIEW_FLOW_GRAPH_TRY_PUT_AND_WAIT
+    template <typename TupleType, typename PortsType>
+    static graph_task* emit_this(graph& g, const TupleType& t, PortsType& p,
```

Comment:
I'm not sure what this method supposed to do. Could you please explain?

Reply:
Sure. This function is a helper function to `split_node`.
Let's say the `split_node` takes a `tuple<Arg1, Arg2, ..., ArgN> and sends the corresponding item to each of N successors.
`split_node::output_ports()` is a tuple of N successors which is passed here (argument `t`).
`emit_this<N>` takes Nth element from the input tuple, calls `try_put_task` on the Nth output port and spawns the returned task. 
New function is just an overload for pre-existing `emit_this` sending `metainfo` together with the element.